pkg/sql/lease.go

```diff
@@ -714,12 +714,10 @@ func (t *tableState) release(lease *LeaseState, store LeaseStore) error {
 	decRefcount := func(s *LeaseState) bool {
 		// Figure out if we'd like to remove the lease from the store asap (i.e. when
 		// the refcount drops to 0). If so, we'll need to mark the lease as released.
-		removeOnceDereferenced := false
-
-		if store.testingKnobs.RemoveOnceDereferenced {
+		removeOnceDereferenced := m.isDrainingLeases()
+		if m.LeaseStore.testingKnobs.RemoveOnceDereferenced {
 			removeOnceDereferenced = true
 		}
```

Comment:
fwiw I think the current form is clearer. if anything, i'd check the testing knob first since it avoids a function call.